test/itkWaveletCoeffsSpatialDomainImageFilterTest.cxx

```diff
@@ -101,64 +101,96 @@ itkWaveletCoeffsSpatialDomainImageFilterTest( int argc, char *argv[] )
   const unsigned int inputLevels = atoi( argv[3] );
   const unsigned int inputBands  = atoi( argv[4] );
   const unsigned int dimension = atoi( argv[6] );
-  constexpr unsigned int ImageDimension = 3;
-  if ( dimension == 2 )
+  constexpr size_t ImageDimension = 3;
+  if ( ! ( dimension == ImageDimension ) )
     {
-    unsigned int ImageDimension = 2;
-    }
-  else if ( dimension == 3 )
-    {
-    unsigned int ImageDimension = 3;
+    std::cerr << "Only 3 dimension supported." << std::endl;
+    std::cerr << "Test failed!" << std::endl;
+    std::cerr << "Error: only 3 dimensions allowed, " << dimension << " selected." << std::endl;
+    return EXIT_FAILURE;
     }
 
   using ImageType = itk::Image<float, ImageDimension>;
 
   using WaveletScalarType = double;
-  using WaveletType = itk::SimoncelliIsotropicWavelet< WaveletScalarType, ImageDimension>;
```

Comment:
@phcerdan Oh.  I forgot I changed this.  

The original code was wrong.  'using' directives are compile-time settings, but the original code was treating it as a runtime variable.  The original code only compiled WaveletCoeffsSpatialDomainImageFilterType with 'WaveletType = itk::SimoncelliIsotropicWavelet'.  All the  other 'using' statementments in the if/else sections were "Shadow types that are unused"

Reply:
I see, thanks @hjmjohnson !